### PR TITLE
[WIP] skip constant folding dequant node

### PR DESCRIPTION
**Pull request by Krovatkin**
This PR makes Constant Propagation to ignore dequant nodes.

https://github.com/pytorch/pytorch/issues/61092

### CODE REVIEW

**Comment by Krovatkin** [comment]
Hey guys, @vkuzo @jerryzh168 I need someone from the quantization team to double check if disabling quant -> dequant folding won't have any unintended consequences. Could you please take a quick look?

**Comment by vkuzo** [comment]
@Krovatkin , thanks for the PR.  Would you mind sharing the motivation for this, what is this fixing?

**Comment by Krovatkin** [comment]
@vkuzo  https://github.com/pytorch/pytorch/issues/61092 describes it in a great detail. Briefly, CP folds quant/dequant nodes before a fusion pass can fuse them at runtime.

**Review by jerryzh168 — APPROVED** [review]
makes senese, we did this for unpack as well, to keep the pattern: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/quantization/quantization_patterns.h#L333, I don't remember how we did it for unpack anymore, if you have time could you also check that to make sure we use the same mechanism for that if possible.

File: test/quantization/jit/test_quantize_jit.py [hunk 1]
```diff
@@ -84,6 +84,50 @@
...
+
+        # After freezing, weight becomes Constant.
+        # We have this pattern in the original graph: Constant f32_weight -> quant -> dequant
+        # After skipping dequant during Constant Propagation, the resulting graph will be:
+        # Constant int8_weight -> dequant
+        FileCheck().check_count("aten::quantize_per_tensor", 2, exactly=True).run(freezed.graph)
+        FileCheck().check_count("aten::quantize_per_channel", 0, exactly=True).run(freezed.graph)
+        FileCheck().check_count("aten::dequantize", 3, exactly=True).run(freezed.graph)
+        FileCheck().check("aten::quantize_per_tensor").check_next("aten::dequantize"
+        ).check_not(
```
Comment (jerryzh168) [inline test/quantization/jit/test_quantize_jit.py:118]: please fix the lint

**Comment by gchanan** [comment]
I'm not sure why this is in the 1.10.1 milestone -- the description is super unclear but it sounds like it's a potential perf improvement, not a bug fix.

**Comment by seemethere** [comment]
Yup I agree with @gchanan, this doesn't appear to fix any regression, going to go ahead and remove this from the 1.10.1 milestone.

**Pull request closed without merging**